ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoPartiesSpec.scala

```diff
@@ -172,33 +158,60 @@ private[dao] trait JdbcLedgerDaoPartiesSpec {
-      party,
-      displayName = Some("Alice Arkwright"),
-      isLocal = true,
-    )
-    val aliceAgain = PartyDetails(
-      party,
-      displayName = Some("Alice Carthwright"),
-      isLocal = true,
-    )
-    val bobDetails = PartyDetails(
-      party = Ref.Party.assertFromString(s"Bob-${UUID.randomUUID()}"),
-      displayName = Some("Bob Bobertson"),
-      isLocal = true,
-    )
-    for {
-      response <- storePartyEntry(aliceDetails, nextOffset())
-      _ = response should be(PersistenceResponse.Ok)
-      response <- storePartyEntry(aliceAgain, nextOffset())
-      _ = response should be(PersistenceResponse.Duplicate)
-      response <- storePartyEntry(bobDetails, nextOffset())
-      _ = response should be(PersistenceResponse.Ok)
-      parties <- ledgerDao.listKnownParties()
-    } yield {
-      parties should contain.allOf(aliceDetails, bobDetails)
+  it should "be able to store multiple parties with the same identifier, and the last update will be visible as query-ing" in {
+    if (enableAppendOnlySchema) {
+      val danParty = Ref.Party.assertFromString(s"Dan-${UUID.randomUUID()}")
+      val dan = PartyDetails(
+        party = danParty,
+        displayName = Some("Dangerous Dan"),
+        isLocal = true,
+      )
+      val dan2 = dan.copy(displayName = Some("Even more dangerous Dan"))
+      val dan3 = dan.copy(displayName = Some("Even more so dangerous Dan"))
+      val dan4 = dan.copy(
+        displayName = Some("Ultimately dangerous Dan"),
+        isLocal = false,
+      )
+      val beforeStartOffset = nextOffset()
+      val firstOffset = nextOffset()
+      for {
+        response <- storePartyEntry(dan, firstOffset)
+        _ = response should be(PersistenceResponse.Ok)
+        secondOffset = nextOffset()
+        response <- storePartyEntry(dan2, secondOffset)
+        _ = response should be(PersistenceResponse.Ok)
+        thirdOffset = nextOffset()
+        response <- storePartyEntry(dan3, thirdOffset)
+        _ = response should be(PersistenceResponse.Ok)
+        lastOffset = nextOffset()
+        response <- storePartyEntry(
+          dan4,
+          lastOffset,
+          Some(Ref.SubmissionId.assertFromString("final submission")),
+        )
+        _ = response should be(PersistenceResponse.Ok)
+        parties <- ledgerDao.getParties(Seq(danParty))
+        partyEntries <- ledgerDao
+          .getPartyEntries(beforeStartOffset, nextOffset())
+          .runWith(Sink.collection)
+      } yield {
+        parties shouldBe List(dan4)
+        val partyEntriesMap = partyEntries.toMap
+        partyEntriesMap(firstOffset).asInstanceOf[AllocationAccepted].partyDetails shouldBe dan
+        partyEntriesMap(secondOffset).asInstanceOf[AllocationAccepted].partyDetails shouldBe dan2
+        partyEntriesMap(thirdOffset).asInstanceOf[AllocationAccepted].partyDetails shouldBe dan3
+        partyEntriesMap(lastOffset).asInstanceOf[AllocationAccepted].partyDetails shouldBe dan4
```

Comment:
Here we would expect that `dan` is still considered to be locally hosted.

Reply:
fixed